### PR TITLE
Add annotation package docs

### PR DESCRIPTION
**Pull request by SimonBaeumer**
Added documentation for the annotation package.

Upstream see: https://github.com/joelanford/helm-operator/pull/92

### CODE REVIEW

File: pkg/annotation/annotation.go [hunk 1]
```diff
@@ -30,27 +54,24 @@ var (
...
 	InstallOption(string) helmclient.InstallOption
 }
 
+// Upgrade configures an upgrade annotation.
 type Upgrade interface {
 	Name() string
 	UpgradeOption(string) helmclient.UpgradeOption
 }
 
+// Uninstall configures an install annotation.
```
Comment (msugakov) [inline pkg/annotation/annotation.go]: Typo: "an **un**install annotation."

File: pkg/annotation/annotation.go [hunk 2]
```diff
@@ -14,6 +14,30 @@ See the License for the specific language governing permissions and
...
+// To disable hooks based on annotations the InstallDisableHooks is passed to the reconciler as an option.
+//
+//    r, err := reconciler.New(
+//	    reconciler.WithChart(*w.Chart),
+//	    reconciler.WithGroupVersionKind(w.GroupVersionKind),
+//	    reconciler.WithInstallAnnotations(annotation.InstallDisableHook{}),
+//	  )
+//
+// If the reconciler detects an annotation named "helm.sdk.operatorframework.io/install-disable-hooks"
+// on the watched custom resource it sets the install.DisableHooks option to the annotations value. For more information
```
Comment (msugakov) [inline pkg/annotation/annotation.go]: Nit, comma is missing: on the watched custom resource**,** it sets ...

File: pkg/annotation/annotation.go [hunk 3]
```diff
@@ -14,6 +14,30 @@ See the License for the specific language governing permissions and
...
+// To create custom annotations implement the Install, Upgrade or Uninstall interface.
+//
+// Example:
+//
+// To disable hooks based on annotations the InstallDisableHooks is passed to the reconciler as an option.
+//
+//    r, err := reconciler.New(
+//	    reconciler.WithChart(*w.Chart),
+//	    reconciler.WithGroupVersionKind(w.GroupVersionKind),
+//	    reconciler.WithInstallAnnotations(annotation.InstallDisableHook{}),
```
Comment (msugakov) [inline pkg/annotation/annotation.go]: Should be `InstallDisableHooks` - plural.

Comment (SimonBaeumer) [reply pkg/annotation/annotation.go]: done

File: pkg/annotation/annotation.go (hunk 2 above)
Comment (SimonBaeumer) [reply pkg/annotation/annotation.go]: done

File: pkg/annotation/annotation.go (hunk 1 above)
Comment (SimonBaeumer) [reply pkg/annotation/annotation.go]: done

**Review by msugakov — APPROVED** [review]

**Pull request merged**